File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ht = {}
        for i in range(len(nums)):
            find = target - nums[i]
            if find in ht:
                return [ht[find], i]
            else:
                ht[nums[i]] = i
        return [-1, -1]

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); ++i) {
            auto it = mp.find(target - nums[i]);
            if (it != mp.end()) {
                return { it->second, i };
            } else {
                mp.insert({nums[i], i});
            }
        }
        return {-1, -1};
    }
};